Common/Orders/Fills/ImmediateFillModel.cs

```diff
@@ -479,5 +536,33 @@ public Prices(decimal current, decimal open, decimal high, decimal low, decimal
                 Close = close == 0 ? current : close;
             }
         }
+
+        private class VwapData
+        {
+            private decimal _sumOfVolume;
+            private DateTime _lastUpdateTime;
+
+            public decimal Vwap { get; private set; }
+
+            public void Update(DateTime time, decimal price, decimal volume)
+            {
+                // this check is needed because fill models are called more than once per time step
+                if (time <= _lastUpdateTime)
+                {
+                    return;
+                }
+
+                _lastUpdateTime = time;
+                _sumOfVolume += volume;
+
+                if (_sumOfVolume == 0)
+                {
+                    return;
+                }
+
+                // update VWAP with current price and volume
+                Vwap += volume / _sumOfVolume * (price - Vwap);
```

Comment:
should make this a computed property, I think we only need the value once at market close, so we don't need to compute this value on every time step.

Reply:
Doesn't the VWAP _need_ to be updated on every time step (every bar/tick) ?

Reply:
didn't realize how you were computing it -- there's two ways to do this, one is the way that's shown here, another is to keep running totals of the sums of dollar volume and volume, and then vwap is $volume/volume, but this works too.